### PR TITLE
[CIAPP] Extend Jenkins documentation for CI Visibility using plugin UI

### PR DESCRIPTION
**Pull request by drodriguezhdez**
### What does this PR do?

This PR extends the Jenkins documentation for CI Visibility using the new plugin UI.

Additionally, the PR adds a `Troubleshooting` section with some common errors.

### Preview

https://docs-staging.datadoghq.com/drodriguezhdez/ciapp_jenkins_docs_improvements

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/setup_pipelines/jenkins.md [hunk 1]
```diff
@@ -84,21 +67,16 @@ With this setup, the Agent listens in port `10518` for logs.
...
-<targetLogCollectionPort>10518</targetLogCollectionPort>
-<collectBuildLogs>true</collectBuildLogs>
-{{< /code-block >}}
-
-3. Restart Jenkins for the changes to take effect.
-
-**Note**: Do not use the Jenkins web GUI to activate logs collection, as trace collection might be disabled when saving settings using the GUI.
+You can correlate your pipeline logs with the traces generated by CI Visibility.
 
+1. In your Jenkins instance web interface, go to **Manage Jenkins > Configure System**.
```
Comment (AdrianLC) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:72]:
```suggestion
1. In the web interface of your Jenkins instance, go to **Manage Jenkins > Configure System**.
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md [hunk 2]
```diff
@@ -159,28 +138,78 @@ datadog(tags: ["team:backend", "release:canary"]){
...
-1. Open the file `$JENKINS_HOME\org.datadog.jenkins.plugins.datadog.DatadogGlobalConfiguration.xml`.
-2. Add or modify the following lines:
-
-    {{< code-block lang="xml" >}}
-    <globalTags>global_tag:global_value,global_tag2:${SOME_ENVVAR},${OTHER_ENVVAR}:global_tagvalue</globalTags>
-    <globalJobTags>(.*?)_job_(*?)_release:someValue</globalJobTags>
-    {{< /code-block >}}
-
-    **Global tags** (`globalTags`): A comma-separated list of tags to apply to all metrics, traces, events, and service checks. Tags can include environment variables that are defined in the Jenkins controller instance.
+1. In your Jenkins instance web interface, go to **Manage Jenkins > Configure System**.
```
Comment (AdrianLC) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:141]: same as above

File: content/en/continuous_integration/setup_pipelines/jenkins.md [hunk 3]
```diff
@@ -16,59 +16,42 @@ Install the [Datadog Agent][1] on the Jenkins controller instance.
...
-When the configuration is correct, the Jenkins log shows lines similar to this after the restart:
-
-{{< code-block lang="text" >}}
-INFO    datadog.trace.core.CoreTracer#<init>: New instance: DDTracer-62fcf62{ ... }
-INFO    datadog.trace.core.StatusLogger#logStatus: DATADOG TRACER CONFIGURATION { ... }
-{{< /code-block >}}
-
-**Note**: Enabling the collection of traces using the Jenkins plugin is incompatible with running the Java APM tracer as a Java agent when launching Jenkins.
-
+4. To verify the plugin is installed, search for `Datadog Plugin` on the **Installed** tab.
```
Comment (AdrianLC) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:24]:
```suggestion
4. To verify that the plugin is installed, search for `Datadog Plugin` on the **Installed** tab.
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md [hunk 4]
```diff
@@ -16,59 +16,42 @@ Install the [Datadog Agent][1] on the Jenkins controller instance.
...
+10. To verify that CI Visibility is enabled, go to `Jenkins Log` and search for:
 
 {{< code-block lang="text" >}}
-INFO    datadog.trace.core.CoreTracer#<init>: New instance: DDTracer-62fcf62{ ... }
-INFO    datadog.trace.core.StatusLogger#logStatus: DATADOG TRACER CONFIGURATION { ... }
+Re/Initialize Datadog-Plugin Tracer: hostname = <HOST>, traceCollectionPort = <TRACE_PORT>
+DATADOG TRACER CONFIGURATION {...}
 {{< /code-block >}}
 
+**Note**: Enabling CI Visibility using the Jenkins plugin is incompatible with running the Java APM tracer as a Java agent when launching Jenkins.
```
Comment (AdrianLC) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:44]:
```suggestion
**Note**: Enabling CI Visibility using the Jenkins plugin is not compatible with running the Java APM tracer as a Java agent when launching Jenkins.
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md [hunk 5]
```diff
@@ -159,28 +138,78 @@ datadog(tags: ["team:backend", "release:canary"]){
...
 
 ## Visualize pipeline data in Datadog
 
 Once the integration is successfully configured, both [Pipelines][6] and [Pipeline Executions][7] pages will start populating with data after pipelines finish.
 
 **Note**: The Pipelines page shows data for only the default branch of each repository.
 
+## Troubleshooting
+
+### Configure DEBUG log level for Datadog Plugin logs
```
Comment (AdrianLC) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:160]:
```suggestion
### Enable DEBUG log level for the Datadog Plugin
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md [hunk 6]
```diff
@@ -159,28 +138,78 @@ datadog(tags: ["team:backend", "release:canary"]){
...
+
+If you have any issue with the Datadog Plugin, you can set the logs for the plugin in a `DEBUG` log level. Using this level, you will be able to check the exception stacktrace if some issue happens.
```
Comment (AdrianLC) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:162]:
```suggestion
If you have any issue with the Datadog Plugin, you can set the logs for the plugin in a `DEBUG` log level. Using this level, you will be able to check the exception stacktrace if there are any issues.
```
or any issue happens, but pretty sure some is wrong here

File: content/en/continuous_integration/setup_pipelines/jenkins.md [hunk 7]
```diff
@@ -159,28 +138,78 @@ datadog(tags: ["team:backend", "release:canary"]){
...
+- Logger: `org.datadog.jenkins.plugins.datadog.clients` -> Log Level `ALL`
+- Logger: `org.datadog.jenkins.plugins.datadog.traces` -> Log Level `ALL`
+- Logger: `org.datadog.jenkins.plugins.datadog.logs` -> Log Level `ALL`
+- Logger: `org.datadog.jenkins.plugins.datadog.model` -> Log Level `ALL`
+- Logger: `org.datadog.jenkins.plugins.datadog.listeners` -> Log Level `ALL`
+5. Save the configuration.
+
+You may also want to split the loggers into different log recorders.
+
+Once the log recorders are successfully configured, you can check the logs in the `DEBUG` mode by accessing to the desired log recorder via **Manage Jenkins > System log**.
```
Comment (AdrianLC) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:177]:
```suggestion
Once the log recorders are successfully configured, you can check the logs in the `DEBUG` mode by accessing the desired log recorder via **Manage Jenkins > System log**.
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md [hunk 8]
```diff
@@ -159,28 +138,78 @@ datadog(tags: ["team:backend", "release:canary"]){
...
+
+If you triggered some Jenkins pipelines, you can search for `Send pipeline traces` message in the **Datadog Plugin Logs**. This message indicates that the plugin is sending **CI Visibility** data to the **Datadog Agent**.
```
Comment (AdrianLC) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:179]:
```suggestion
If you trigger a Jenkins pipelines, you can search for the message `Send pipeline traces` in the **Datadog Plugin Logs**. This message indicates that the plugin is sending **CI Visibility** data to the **Datadog Agent**.
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md [hunk 9]
```diff
@@ -159,28 +138,78 @@ datadog(tags: ["team:backend", "release:canary"]){
...
+{{< /code-block >}}
+
+1. Access to your Jenkins controller instance machine via console.
+2. Check the Java startup line for the Jenkins process.
+{{< code-block lang="shell" >}}
+$> ps -ef | grep jenkins
+{{< /code-block >}}
+3. Ensure you are not using the APM Java Tracer as `-javaagent`.
+
+**Note**: Enabling CI Visibility using the Jenkins plugin is incompatible with running the Java APM tracer as a Java agent when launching Jenkins.
```
Comment (AdrianLC) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:212]: not compatible

**Review by AdrianLC — APPROVED** [review]
I added some suggestions to the phrasing but 👍

File: content/en/continuous_integration/setup_pipelines/jenkins.md [hunk 10]
```diff
@@ -16,59 +16,42 @@ Install the [Datadog Agent][1] on the Jenkins controller instance.
...
-{{< /code-block >}}
-
-3. Restart Jenkins for the changes to take effect.
-
-If the configuration is correct, your Jenkins log should contain the following lines after restarting:
+1. In your Jenkins instance web interface, go to **Manage Jenkins > Configure System**.
+2. Go to the `Datadog Plugin` section, scrolling down the configuration screen.
+3. Select the `Datadog Agent` mode.
+4. Configure the `Datadog Agent` host.
+5. Configure the `Traces Collection` port. (By default `8126` in the Datadog Agent)
```
Comment (cswatt) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:32]:
```suggestion
5. Configure the `Traces Collection` port. (The default port is `8126` in the Datadog Agent.)
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md [hunk 11]
```diff
@@ -159,28 +138,78 @@ datadog(tags: ["team:backend", "release:canary"]){
...
+5. Configure the `Global Job Tags`.
+6. Save your configuration.
 
-    **Global job tags** (`globalJobTags`): A comma-separated list of regex to match a job and a list of tags to apply to that job. Tags can include environment variables that are defined in the Jenkins controller instance. Tags can reference match groups in the regex using the `$` symbol.
+**Global tags** (`globalTags`): A comma-separated list of tags to apply to all metrics, traces, events, and service checks. Tags can include environment variables that are defined in the Jenkins controller instance: `global_tag:global_value,global_tag2:${SOME_ENVVAR},${OTHER_ENVVAR}:global_tagvalue`
 
-3. Restart Jenkins for the changes to take effect.
-
-**Note**: Do not use the Jenkins GUI to configure global tags, as trace collection might be disabled when saving settings using the GUI.
+**Global job tags** (`globalJobTags`): A comma-separated list of regex to match a job and a list of tags to apply to that job. Tags can include environment variables that are defined in the Jenkins controller instance. Tags can reference match groups in the regex using the `$` symbol: `(.*?)_job_(*?)_release:someValue`
```
Comment (cswatt) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:150]:
```suggestion
**Global job tags** (`globalJobTags`): A comma-separated list of regexes to match a job and a list of tags to apply to that job. Tags can include environment variables that are defined in the Jenkins controller instance. Tags can reference match groups in the regex using the `$` symbol: `(.*?)_job_(*?)_release:someValue`
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md (hunk 6 above)
Comment (cswatt) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:162]:
```suggestion
If you have any issues with the Datadog Plugin, you can set the logs for the plugin in a `DEBUG` log level. Using this level, you are able to check the exception stacktrace if an issue arises.
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md (hunk 7 above)
Comment (cswatt) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:177]:
```suggestion
Once the log recorders are successfully configured, you can check the logs in the `DEBUG` mode by accessing the desired log recorder through **Manage Jenkins > System log**.
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md [hunk 12]
```diff
@@ -159,28 +138,78 @@ datadog(tags: ["team:backend", "release:canary"]){
...
+{{< code-block lang="text" >}}
+Send pipeline traces.
+...
+Send pipeline traces.
+...
+{{< /code-block >}}
+
+### The Datadog Plugin section does not appear in the Jenkins configuration
+
+If the Datadog Plugin section does not appear in Jenkins configuration section, you need to make sure that the plugin is enabled. To do so:
```
Comment (cswatt) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:190]:
```suggestion
If the Datadog Plugin section does not appear in Jenkins configuration section, make sure that the plugin is enabled. To do so:
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md [hunk 13]
```diff
@@ -159,28 +138,78 @@ datadog(tags: ["team:backend", "release:canary"]){
...
+
+1. In your Jenkins instance web interface, go to **Manage Jenkins > Manage Plugins**.
+2. Search for `Datadog Plugin` in the **Installed** tab.
+3. Check that the `Enabled` checkbox is marked.
+4. If you enable the plugin here, restart your Jenkins instance using `/safeRestart` URL path.
```
Comment (cswatt) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:195]:
```suggestion
4. If you enable the plugin here, restart your Jenkins instance using the `/safeRestart` URL path.
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md [hunk 14]
```diff
@@ -159,28 +138,78 @@ datadog(tags: ["team:backend", "release:canary"]){
...
+
+### The Plugin Tracer failed to initialized due to APM Java Tracer is used to instrument Jenkins.
+
+If this error message appears in the **Jenkins Log**, you need to make sure that you are not using the APM Java Tracer to instrument your Jenkins instance.
```
Comment (cswatt) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:199]:
```suggestion
If this error message appears in the **Jenkins Log**, make sure that you are not using the APM Java Tracer to instrument your Jenkins instance.
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md [hunk 15]
```diff
@@ -159,28 +138,78 @@ datadog(tags: ["team:backend", "release:canary"]){
...
+
+{{< code-block lang="text" >}}
+Failed to reinitialize Datadog-Plugin Tracer, Cannot enable traces collection via plugin if the Datadog Java Tracer is being used as javaagent in the Jenkins startup command. This error will not affect your pipelines executions.
+{{< /code-block >}}
+
+1. Access to your Jenkins controller instance machine via console.
```
Comment (cswatt) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:205]:
```suggestion
1. Access your Jenkins controller instance machine using the console.
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md (hunk 2 above)
Comment (cswatt) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:141]:
```suggestion
1. In the web interface of your Jenkins instance, go to **Manage Jenkins > Configure System**.
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md (hunk 8 above)
Comment (cswatt) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:179]:
```suggestion
If you trigger a Jenkins pipeline, you can search for the message `Send pipeline traces` in the **Datadog Plugin Logs**. This message indicates that the plugin is sending **CI Visibility** data to the **Datadog Agent**.
```

File: content/en/continuous_integration/setup_pipelines/jenkins.md (hunk 9 above)
Comment (cswatt) [inline content/en/continuous_integration/setup_pipelines/jenkins.md:212]:
```suggestion
**Note**: Enabling CI Visibility using the Jenkins plugin is not compatible with running the Java APM tracer as a Java agent when launching Jenkins.
```

**Review by cswatt — APPROVED** [review]

**Pull request merged**